### PR TITLE
Revert via

### PR DESCRIPTION
**Pull request by zwoop**
This reverts #1988 and #2044.

### CODE REVIEW

File: proxy/InkAPI.cc [hunk 1]
```diff
@@ -8431,6 +8409,7 @@ TSHttpTxnConfigFind(const char *name, int length, TSOverridableConfigKey *conf,
         cnf = TS_CONFIG_HTTP_FLOW_CONTROL_ENABLED;
       }
       break;
+      break;
```
Comment (SolidWallOfCode) [inline proxy/InkAPI.cc]: Don't need 2 of these.

File: lib/records/I_RecCore.h [hunk 2]
```diff
@@ -176,7 +176,6 @@ int RecGetRecordBool(const char *name, RecBool *rec_byte, bool lock = true);
 typedef void (*RecLookupCallback)(const RecRecord *, void *);
 
 int RecLookupRecord(const char *name, RecLookupCallback callback, void *data, bool lock = true);
-int RecLookupRecord(const char *name, std::function<void(const RecRecord *)> &&f, bool lock = true);
```
Comment (SolidWallOfCode) [inline lib/records/I_RecCore.h]: I may bring this back in another PR, I think it's useful in general.

File: lib/records/RecCore.cc [hunk 3]
```diff
@@ -484,25 +484,6 @@ RecLookupRecord(const char *name, void (*callback)(const RecRecord *, void *), v
   return err;
 }
 
-namespace
```
Comment (SolidWallOfCode) [inline lib/records/RecCore.cc]: This too, related to the previous comment.

File: mgmt/RecordsConfig.cc [hunk 4]
```diff
@@ -362,17 +362,12 @@ static const RecordElement RecordsConfig[] =
   //        # verbose via string
   //        #
   //        # 0 - no extra info added to string
-  //        # 1 - minimal information.
-  //        # 2 - more information
-  //        # 3 - maximum information.
```
Comment (SolidWallOfCode) [inline mgmt/RecordsConfig.cc]: Might want to keep this, still accurate.

File: proxy/http/HttpConfig.cc [hunk 5]
```diff
@@ -67,18 +65,16 @@ class HttpConfigCont : public Continuation
 /// Data item for enumerated type config value.
 template <typename T> struct ConfigEnumPair {
```
Comment (SolidWallOfCode) [inline proxy/http/HttpConfig.cc]: Should probably remove all of `ConfigEnumPair`.

File: proxy/http/HttpConfig.cc [hunk 6]
```diff
@@ -67,18 +65,16 @@ class HttpConfigCont : public Continuation
...
-  ConfigEnumPair(T v, char const *k) : _value(v), _key(k) {}
+  const char *_key;
 };
 
 /// Convert a string to an enumeration value.
 /// @a n is the number of entries in the list.
 /// @return @c true if the string is found, @c false if not found.
 /// If found @a value is set to the corresponding value in @a list.
-template <typename E, unsigned N>
+template <typename T, unsigned N>
```
Comment (SolidWallOfCode) [inline proxy/http/HttpConfig.cc]: Basically all of the enum support for `records.config` should be removed.

File: proxy/http/HttpTransactHeaders.cc [hunk 7]
```diff
@@ -683,82 +683,24 @@ HttpTransactHeaders::insert_server_header_in_response(const char *server_tag, in
...
-      *via++ = ' ';
-    }
-  }
-  return via - via_string;
-}
-
 /// Look up the protocol stack and write it to the @a via_string.
 size_t
-write_client_protocol_stack(char *via_string, size_t len, HttpTransact::State *s)
+write_client_protocol_stack(HttpTransact::State *s, char *via_string, size_t len)
```
Comment (SolidWallOfCode) [inline proxy/http/HttpTransactHeaders.cc]: This needs to be removed and reverted. I would grab the entire file from before #1534 or from #2018.

File: tests/gold_tests/headers/via.gold [hunk 8]
```diff
@@ -2,5 +2,3 @@ Via: http/1.1-tcp-ipv4
 Via: http/1.0-tcp-ipv4
```
Comment (SolidWallOfCode) [inline tests/gold_tests/headers/via.gold]: This entire file needs removed.

File: tests/gold_tests/headers/via.test.py [hunk 9]
```diff
@@ -26,10 +26,8 @@
 '''
```
Comment (SolidWallOfCode) [inline tests/gold_tests/headers/via.test.py]: This file needs to be removed.

**Comment by zwoop** [comment]
Please make your own revert, I only reverted the commits as is. I'd like to get this and the necessary 7.1.0 revert I'm ad soon as possible, so we can get a release out at some point....

**Pull request closed without merging**